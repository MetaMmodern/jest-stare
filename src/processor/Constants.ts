import * as chalk from "chalk";

/**
 * Contains Constants constants
 * @export
 * @class Constants
 */
export class Constants {

    /**
     * Name (same as package.json)
     * @static
     * @memberof Constants
     */
    public static readonly NAME = "jest-stare";

    /**
     * Logo for messaging
     * @static
     * @memberof Constants
     */
    public static readonly LOGO = chalk.default.green("**  ") + chalk.default.green("jest") + chalk.default.yellow("-") + chalk.default.red("stare");

    /**
     * Default jest-stare results directory
     * @static
     * @memberof Constants
     */
    public static readonly DEFAULT_RESULTS_DIR = "./" + Constants.NAME;

    /**
     * Main HTML index file for report
     * @static
     * @memberof Constants
     */
    public static readonly MAIN_HTML = "index.html";

    /**
     * jest-stare render JS file
     * @static
     * @memberof Constants
     */
    public static readonly JEST_STARE_JS = "view.js";

    /**
     * Raw JSON response results
     * @static
     * @memberof Constants
     */
    public static readonly RESULTS_RAW = "jest-results.json";

    /**
     * jest-stare custom css
     * @static
     * @memberof Constants
     */
    public static readonly JEST_STARE_CSS = Constants.NAME + ".css";

    /**
     * Template HTML file
     * @static
     * @memberof Constants
     */
    public static readonly TEMPLATE_HTML = "template.html";

    /**
     * Base dir to contain css files
     * @static
     * @memberof Constants
     */
    public static readonly CSS_DIR = "css/";

    /**
     * Base dir to contain js files
     * @static
     * @memberof Constants
     */
    public static readonly JS_DIR = "js/";

    /**
     * Log content after logo
     * @static
     * @memberof Constants
     */
    public static readonly LOG_MESSAGE = " --testResultsProcessor: wrote output report to ";
}
