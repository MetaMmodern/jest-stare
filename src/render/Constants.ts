/**
 * Constants for rendering
 * @export
 * @class Constants
 */
export class Constants {

    /**
     * Label for passed tests
     * @static
     * @memberof Constants
     */
    public static readonly PASSED_LABEL = "Passed";

    /**
     * Label for failed tests
     * @static
     * @memberof Constants
     */
    public static readonly FAILED_LABEL = "Failed";

    /**
     * Passed jest status
     * @static
     * @memberof Constants
     */
    public static readonly TEST_STATUS_PASS = "passed";

    /**
     * Failed jest status
     * @static
     * @memberof Constants
     */
    public static readonly TEST_STATUS_FAIL = "failed";

    /**
     * Pending jest status
     * @static
     * @memberof Constants
     */
    public static readonly TEST_STATUS_PEND = "pending";

    /**
     * Obsolete color
     * @static
     * @memberof Constants
     */
    public static readonly OBSOLETE = "xx";

    /**
     * Pass color
     * @static
     * @memberof Constants
     */
    public static readonly PASS_RAW = "28a745";
    public static readonly PASS = "#" + Constants.PASS_RAW;

    /**
     * Fail color
     * @static
     * @memberof Constants
     */
    public static readonly FAIL_RAW = "dc3545";
    public static readonly FAIL = "#" + Constants.FAIL_RAW;

    /**
     * Passed test class
     * @static
     * @memberof Constants
     */
    public static readonly PASSED_TEST = "passed-test";

    /**
     * Failed test class
     * @static
     * @memberof Constants
     */
    public static readonly FAILED_TEST = "failed-test";

    /**
     * Both passed and failed test class
     * @static
     * @memberof Constants
     */
    public static readonly BOTH_TEST = "both-test";
}
