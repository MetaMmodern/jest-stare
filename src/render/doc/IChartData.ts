
/**
 * Holds chart data
 * @export
 * @interface IChartData
 */
export interface IChartData {

    /**
     * Label for same index data
     * @type {string[]}
     * @memberof IChartData
     */
    labels: string[];

    /**
     * Color for same index data
     * @type {string[]}
     * @memberof IChartData
     */
    backgroundColor: string[];

    /**
     * Raw data values
     * @type {number[]}
     * @memberof IChartData
     */
    data: number[];
}
