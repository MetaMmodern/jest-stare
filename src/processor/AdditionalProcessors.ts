import { IResultsProcessorInput } from "./doc/jest/IResultsProcessorInput";
import { Logger } from "../utils/Logger";
import { Constants } from "./Constants";


export class AdditionalProcessors {

    /**
     * Pass the result processor input given to jest-stare to additional
     * test results processors
     * @param jestTestData - input passed to jest-stare
     * @param processors - list of test results processors (e.g. ["jest-html-reporter"])
     *                     to forward the data to
     */
    public static execute(jestTestData: IResultsProcessorInput, processors: string[]): void {
        const chalk = require("chalk");
        for (const processor of processors) {
            try {
                require(processor)(jestTestData);
                AdditionalProcessors.logger.debug(Constants.LOGO + " passed results to additional processor \"" +
                    processor + "\"" + chalk.default.green("\t**"));
            } catch (e) {
                AdditionalProcessors.logger.error("Error executing additional processor: \"" + processor + "\"" + e);
            }
        }
    }

    private static logger = new Logger();
}
