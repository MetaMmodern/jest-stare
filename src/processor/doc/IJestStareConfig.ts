import { Constants } from "../Constants";

/**
 * Interface mapping fields the user can specify in package.json
 * @export
 * @interface IJestStareConfig
 */
export interface IJestStareConfig {

    /**
     * The directory you would like the results to go to
     * @type {string}
     * @memberof IJestStareConfig
     */
    resultDir?: string;

    /**
     * Configure whether logging is on (true) or off (false)
     * @type {boolean}
     * @memberof IJestStareConfig
     */
    log?: boolean;

    /**
     * An array of additional package names to use as testResultsProcessors
     * This way you can produce other reports from the same data
     * using other packages such as jest-html-reporter
     * @type {string[]}
     * @memberof IJestStareConfig
     */
    additionalResultsProcessors?: string[];
}

/**
 * Key in package.json that specified jest-stare config
 */
export const PACKAGE_JSON_KEY = Constants.NAME;
