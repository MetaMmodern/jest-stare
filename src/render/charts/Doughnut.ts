import { Constants } from "../Constants";
import { Chart, ChartConfiguration } from "chart.js";

/**
 * Handles creating doughnut charts
 * @export
 * @class Doughnut
 */
export class Doughnut {

    /**
     * Generate charts from input tag prefx
     * @static
     * @param {JQuery<HTMLCanvasElement>} canvas - canvas for chart to add to
     * @param {string[]} labels - labels for data
     * @param {string[]} backgroundColor - background colors for data
     * @param {number[]} data - data values
     * @memberof Doughnut
     */
    public static createChart(canvas: JQuery<HTMLCanvasElement>, labels: string[], backgroundColor: string[], data: number[]) {

        const doughnut = "doughnut";

        const config: ChartConfiguration = {
            type: doughnut,
            data: {
                labels,
                datasets: [
                    {
                        backgroundColor,
                        data,
                    }
                ]
            }
        };

        Doughnut.buildCanvas(canvas.get(0), config);
    }

    /**
     * Build charts at a canvas with input config
     * @static
     * @param {HTMLCanvasElement} canvas - canvas location
     * @param {ChartConfiguration} config - config for chart
     * @memberof Doughnut
     */
    public static buildCanvas(canvas: HTMLCanvasElement, config: ChartConfiguration) {
        const doughnut = new Chart(canvas, config);
    }

}
