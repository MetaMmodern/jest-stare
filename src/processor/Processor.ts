import { Constants } from "./Constants";
import { IResultsProcessorInput } from "./doc/jest/IResultsProcessorInput";
import { ISubstitute } from "./doc/ISubstitute";
import { IO } from "../utils/IO";
import * as mustache from "mustache";
import * as path from "path";
import { IJestStareConfig, PACKAGE_JSON_KEY } from "./doc/IJestStareConfig";
import { Logger } from "../utils/Logger";
import * as chalk from "chalk";
import { IThirdPartyDependency } from "./doc/IThirdPartyDependency";
import { Dependencies } from "./Dependencies";
import { isNullOrUndefined } from "util";
import { IProcessParms } from "./doc/IProcessParms";
import { EnvironmentalVariables } from "../utils/EnvironmentalVariables";
const pkgUp = require("pkg-up");

/**
 * Class to post process jest output and summarize information in an html file
 * @export
 * @class Processor
 */
export class Processor {

    /**
     * Main exported method to obtain and return summary results
     * @static
     * @param {IResultsProcessorInput} results - input results object
     * @param {IJestStareConfig} [explicitConfig] - programmatic config
     * @param {IProcessParms} [mProcessParms] - parms object to control process behavior
     * @returns - returns input results object
     * @memberof Processor
     */
    public static run(results: IResultsProcessorInput, explicitConfig?: IJestStareConfig,
                      parms?: IProcessParms) {

        return new Processor(results, explicitConfig, parms).generate();
    }

    /**
     * Instance of our logger
     * @private
     * @type {Logger}
     * @memberof Processor
     */
    private mLog: Logger;

    /**
     * Creates an instance of Processor.
     * @param {IResultsProcessorInput} results - input results object
     * @param {IJestStareConfig} [explicitConfig] - programmatic config
     * @param {IProcessParms} [mProcessParms] - parms object to control process behavior
     * @memberof Processor
     */
    constructor(private mResults: IResultsProcessorInput, private mExplicitConfig?: IJestStareConfig,
                private mProcessParms?: IProcessParms) {
    }

    /**
     * Generate a report after constructed
     * @private
     * @returns
     * @memberof Processor
     */
    private generate() {
        const substitute: ISubstitute = {};

        // throw error if no input object
        if (isNullOrUndefined(this.mResults)) {
            throw new Error(Constants.NO_INPUT);
        }

        const config = this.buildConfig();

        // build mustache render substitution values
        substitute.results = this.mResults;
        substitute.rawResults = JSON.stringify(this.mResults, null, 2);
        substitute.jestStareConfig = config;
        substitute.rawJestStareConfig = JSON.stringify(config, null, 2);

        // save in reporter
        if (this.mProcessParms && this.mProcessParms.reporter) {
            this.mProcessParms.reporter.jestStareConfig = config;
        }

        // generate report
        this.generateReport(config.resultDir, substitute, this.mProcessParms);

        if (config.additionalResultsProcessors != null) {
            this.execute(this.mResults, config.additionalResultsProcessors);
        }
        // return back to jest
        return this.mResults;
    }

    /**
     * Build config from explicit config, package.json, and defaults
     * @private
     * @returns {IJestStareConfig} - constructed config
     * @memberof Processor
     */
    private buildConfig(): IJestStareConfig {

        // get configuration
        const packageJsonConfig = this.readPackageJson();

        // read environmental variables and merge them with the package.json config (env takes precedence)
        const envVars = new EnvironmentalVariables();
        const mergedEnvAndPackageJsonConfig = envVars.resolve(packageJsonConfig, envVars.read());

        // explicit config takes precedence over  env and package.json
        const config = this.mExplicitConfig || mergedEnvAndPackageJsonConfig;

        // take packagejson options after setting explicit config (concatenate both)
        if (this.mExplicitConfig != null) {
            Object.keys(mergedEnvAndPackageJsonConfig).forEach((key) => {
                if (isNullOrUndefined(this.mExplicitConfig[key]) && !isNullOrUndefined(mergedEnvAndPackageJsonConfig[key])) {
                    config[key] = mergedEnvAndPackageJsonConfig[key];
                }
            });
        }

        if (config.resultDir == null) {
            config.resultDir = Constants.DEFAULT_RESULTS_DIR;
        }

        // suppress logging if requested
        // NOTE(Kelosky): must be first, to suppress all logging
        if (!isNullOrUndefined(config.log)) {
            if (!config.log) {
                this.logger.on = false;
            }
        }

        // record if we were invoked programmatically
        // NOTE(Kelosky): should be second, to record if override config
        if (!isNullOrUndefined(this.mExplicitConfig)) {

            // display if not internal invocation
            if (this.mProcessParms && this.mProcessParms.reporter) {
                // do nothing
            } else {
                this.logger.info(Constants.OVERRIDE_JEST_STARE_CONFIG);
            }
        }

        if (isNullOrUndefined(config.resultHtml)) {
            this.logger.debug("Setting to default resultHtml");
            config.resultHtml = Constants.MAIN_HTML;
        } else {
            if (config.resultHtml.indexOf( Constants.HTML_EXTENSION) === -1){
                // add .html if the user did not specify it
                config.resultHtml = config.resultHtml + Constants.HTML_EXTENSION;
            }
        }

        if (isNullOrUndefined(config.resultJson)) {
            config.resultJson = Constants.RESULTS_RAW;
        }

        return config;
    }

    /**
     * Create HTML report
     * @private
     * @param {string} resultDir -  directory to save report
     * @param {ISubstitute} substitute - substitution values for mustache render
     * @param {ISettings} settings - settings for IO
     * @memberof Processor
     */
    private generateReport(resultDir: string, substitute: ISubstitute, parms: IProcessParms) {

        // create base html file
        resultDir = resultDir + "/"; // append an extra slash in case the user didn't add one
        IO.mkdirsSync(resultDir);
        IO.writeFileSync(resultDir + substitute.jestStareConfig.resultHtml,
            mustache.render(this.obtainWebFile(Constants.TEMPLATE_HTML), substitute));

        // create raw json
        IO.writeFileSync(resultDir + substitute.jestStareConfig.resultJson, substitute.rawResults);

        // create jest-stare config if requested
        if (!isNullOrUndefined(substitute.jestStareConfig.jestStareConfigJson)) {
            IO.writeFileSync(resultDir + substitute.jestStareConfig.jestStareConfigJson, substitute.rawJestStareConfig);
        }

        // create our css
        const cssDir = resultDir + Constants.CSS_DIR;
        IO.mkdirsSync(cssDir);
        IO.writeFileSync(cssDir + Constants.JEST_STARE_CSS, this.obtainWebFile(Constants.JEST_STARE_CSS));

        // create our js
        const jsDir = resultDir + Constants.JS_DIR;
        IO.mkdirsSync(jsDir);
        IO.writeFileSync(jsDir + Constants.JEST_STARE_JS, this.obtainJsRenderFile(Constants.JEST_STARE_JS));

        // add third party dependencies
        Dependencies.THIRD_PARTY_DEPENDENCIES.forEach((dependency) => {
            // dependency.targetDir = resultDir + dependency.targetDir;
            const updatedDependency = Object.assign({}, ...[dependency]);
            updatedDependency.targetDir = resultDir + dependency.targetDir;
            this.addThirdParty(updatedDependency);
        });

        // log complete
        let type = " ";
        type += (parms && parms.reporter) ? Constants.REPORTERS : Constants.TEST_RESULTS_PROCESSOR;
        this.logger.info(Constants.LOGO + type + Constants.LOG_MESSAGE + resultDir + substitute.jestStareConfig.resultHtml + Constants.SUFFIX);
    }

    /**
     * Pass the result processor input given to jest-stare to additional
     * test results processors
     * @param jestTestData - input passed to jest-stare
     * @param {IResultsProcessorInput} jestTestData - input passed to jest-stare
     * @param {string[]} processors - processors
     * @param processors - list of test results processors (e.g. ["jest-html-reporter"])
     *                     to forward the data to
     * @memberof Processor
     */
    private execute(jestTestData: IResultsProcessorInput, processors: string[]): void {
        for (const processor of processors) {
            if (processor === Constants.NAME) {
                this.logger.error("Error: In order to avoid infinite loops, " +
                    "jest-stare cannot be listed as an additional processor. Skipping... ");
                continue;
            }
            try {
                require(processor)(jestTestData);
                this.logger.info(Constants.LOGO + " passed results to additional processor " +
                    chalk.default.white("\"" + processor + "\"") + Constants.SUFFIX);
            } catch (e) {
                this.logger.error("Error executing additional processor: \"" + processor + "\" " + e);
            }
        }
    }

    /**
     * Add all third party dependencies
     * @private
     * @param {IThirdPartyDependency} dependency - a dependency to add
     * @memberof Processor
     */
    private async addThirdParty(dependency: IThirdPartyDependency) {
        const location = require.resolve(dependency.requireDir + dependency.file);
        await IO.writeFileSync(dependency.targetDir + dependency.file, IO.readFileSync(location));
    }

    /**
     * Obtain web files
     * @private
     * @returns {string} - file contents from web directory
     * @memberof Processor
     */
    private obtainWebFile(name: string): string {
        return IO.readFileSync(path.resolve(__dirname + "/../../web/" + name));
    }

    /**
     * Obtain js files
     * @private
     * @returns {string} - js file contents from js directory
     * @memberof Processor
     */
    private obtainJsRenderFile(name: string): string {
        return IO.readFileSync(path.resolve(__dirname + "/../render/" + name));
    }

    /**
     * Read from the user's package.json, if present
     * @private
     * @returns {IJestStareConfig} - config object
     * @memberof Processor
     */
    private readPackageJson(): IJestStareConfig {
        const packageJson = pkgUp.sync();
        if (packageJson !== null) {
            const packageJsonContents = IO.readFileSync(packageJson).toString();
            const packageJsonObject = JSON.parse(packageJsonContents);
            if (packageJsonObject[PACKAGE_JSON_KEY] == null) {
                // package json found, but no jest stare config
                return {};
            } else {
                // found the user's package.json config
                return packageJsonObject[PACKAGE_JSON_KEY];
            }
        } else {
            // if we can't find any package.json, return a blank config
            return {};
        }

    }

    /**
     * Set logger instance
     * @private
     * @memberof Processor
     */
    set logger(logger: Logger) {
        this.mLog = logger;
    }

    /**
     * Get log instance
     * @readonly
     * @private
     * @memberof Processor
     */
    get logger() {
        if (isNullOrUndefined(this.mLog)) {
            this.logger = new Logger();
        }
        // console.log((this.mLog as any).writeStdout)
        return this.mLog;
    }
}
