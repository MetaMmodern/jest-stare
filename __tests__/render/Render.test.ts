import * as $ from "jquery";
import { Render } from "../../src/render/Render";
import { Processor } from "../../src/processor/Processor";
import { Constants } from "../../src/processor/Constants";
import { IChartData } from "../../src/render/doc/IChartData";
import { IJestStareConfig } from "../../src/processor/doc/IJestStareConfig";
import { Doughnut } from "../../src/render/charts/Doughnut";

const simplePassingTests: jest.AggregatedResult = require("../__resources__/simplePassingTests.json");
const simpleFailingTests: jest.AggregatedResult = require("../__resources__/simpleFailingTests.json");
const pendingOnlyTests: jest.AggregatedResult = require("../__resources__/pendingOnlyTests.json");

describe("Render tests", () => {

    it("should show snapshots area if snapshots", () => {
        const template = (new Processor(undefined) as any).obtainWebFile(Constants.TEMPLATE_HTML);
        document.write(template);

        (Render as any).updateStatusArea(simplePassingTests);
        const div = $("#snapshots-group") as JQuery<HTMLDivElement>;
        expect(div.css("display")).toBe("block");
    });

    it("should not show snapshots area if no snapshots", () => {
        const template = (new Processor(undefined) as any).obtainWebFile(Constants.TEMPLATE_HTML);
        document.write(template);

        (Render as any).updateStatusArea(pendingOnlyTests);
        const div = $("#snapshots-group") as JQuery<HTMLDivElement>;
        const p = $("#snapshots-results") as JQuery<HTMLParagraphElement>;
        expect(div.css("display")).toBe("none");
    });

    it("should not create link to coverage report if not in config", () => {

        const template = (new Processor(undefined) as any).obtainWebFile(Constants.TEMPLATE_HTML);
        document.write(template);

        const config: IJestStareConfig = {
        };

        (Render as any).setCoverageLink(config);
        expect($("#coverage-link").attr("href")).toMatchSnapshot();
        expect($("#coverage-link").hasClass("disabled")).toBe(true);
    });

    it("should create link to coverage report if in config", () => {

        /**
         * Need css & js collocated and to use document.write to invoke scripts
         */
        const template = (new Processor(undefined) as any).obtainWebFile(Constants.TEMPLATE_HTML);
        document.write(template);

        const config: IJestStareConfig = {
            coverageLink: "/some/place/index.html",
        };

        $("#test-config").text(JSON.stringify(config, null, 2)); // this isn't needed but keeping for future reference
        (Render as any).setCoverageLink(config);
        expect($("#coverage-link").attr("href")).toMatchSnapshot();
        expect($("#coverage-link").hasClass("active")).toBe(true);
    });

    it("should build chart data with no examples", () => {
        expect((Render as any).buildChartsData(0, 0)).toMatchSnapshot();
    });

    it("should build chart data with fail examples", () => {
        expect((Render as any).buildChartsData(0, 1)).toMatchSnapshot();
    });

    it("should build chart data with pass examples", () => {
        expect((Render as any).buildChartsData(1, 0)).toMatchSnapshot();
    });

    it("should build chart data with passing, pending, and failing examples", () => {
        expect((Render as any).buildChartsData(1, 2, 1)).toMatchSnapshot();
    });

    it("should build chart data with pending and fail examples only", () => {
        expect((Render as any).buildChartsData(0, 1, 1)).toMatchSnapshot();
    });

    it("should build chart data with pass and fail examples", () => {
        expect((Render as any).buildChartsData(1, 2)).toMatchSnapshot();
    });

    it("should add snapshot chart data with no examples", () => {
        const chartData: IChartData = (Render as any).buildChartsData(0, 0);
        expect((Render as any).addSnapshotChartData(simplePassingTests, chartData)).toMatchSnapshot();
    });

    it("should add snapshot chart data with pass examples", () => {
        const chartData: IChartData = (Render as any).buildChartsData(2, 0);
        expect((Render as any).addSnapshotChartData(simplePassingTests, chartData)).toMatchSnapshot();
    });

    it("should add snapshot chart data with fail examples", () => {
        const chartData: IChartData = (Render as any).buildChartsData(0, 2);
        expect((Render as any).addSnapshotChartData(simpleFailingTests, chartData)).toMatchSnapshot();
    });

    it("should not create donut charts if config has charts disabled", ()=>{
        const results = require("../__resources__/simplePassingTests.json");
        Doughnut.createChart = jest.fn(()=>{
            throw new Error("Should not have been called.");
        });
        (Render as any).show(results, { disableCharts: true});
        expect(Doughnut.createChart).not.toBeCalled();
    });

    it("should create donut charts if config has charts enabled", ()=>{
        const results = require("../__resources__/simplePassingTests.json");
        Doughnut.createChart = jest.fn(()=>{
            // should be called
        });
        (Render as any).show(results, { disableCharts: false});
        expect(Doughnut.createChart).toBeCalled();
    });


});
