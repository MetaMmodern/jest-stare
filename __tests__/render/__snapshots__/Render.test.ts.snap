// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render tests should add snapshot chart data with fail examples 1`] = `
Object {
  "backgroundColor": Array [
    "#dc3545",
  ],
  "data": Array [
    2,
  ],
  "labels": Array [
    "Failed",
  ],
}
`;

exports[`Render tests should add snapshot chart data with no examples 1`] = `
Object {
  "backgroundColor": Array [
    "#007bff",
  ],
  "data": Array [
    "1",
  ],
  "labels": Array [
    "Added",
  ],
}
`;

exports[`Render tests should add snapshot chart data with pass examples 1`] = `
Object {
  "backgroundColor": Array [
    "#28a745",
    "#007bff",
  ],
  "data": Array [
    2,
    "1",
  ],
  "labels": Array [
    "Passed",
    "Added",
  ],
}
`;

exports[`Render tests should build chart data with fail examples 1`] = `
Object {
  "backgroundColor": Array [
    "#dc3545",
  ],
  "data": Array [
    1,
  ],
  "labels": Array [
    "Failed",
  ],
}
`;

exports[`Render tests should build chart data with no examples 1`] = `
Object {
  "backgroundColor": Array [],
  "data": Array [],
  "labels": Array [],
}
`;

exports[`Render tests should build chart data with pass and fail examples 1`] = `
Object {
  "backgroundColor": Array [
    "#28a745",
    "#dc3545",
  ],
  "data": Array [
    1,
    2,
  ],
  "labels": Array [
    "Passed",
    "Failed",
  ],
}
`;

exports[`Render tests should build chart data with pass examples 1`] = `
Object {
  "backgroundColor": Array [
    "#28a745",
  ],
  "data": Array [
    1,
  ],
  "labels": Array [
    "Passed",
  ],
}
`;

exports[`Render tests should build chart data with passing, pending, and failing examples 1`] = `
Object {
  "backgroundColor": Array [
    "#28a745",
    "#dc3545",
    "#ffc107",
  ],
  "data": Array [
    1,
    2,
    1,
  ],
  "labels": Array [
    "Passed",
    "Failed",
    "Pending",
  ],
}
`;

exports[`Render tests should build chart data with pending and fail examples only 1`] = `
Object {
  "backgroundColor": Array [
    "#dc3545",
    "#ffc107",
  ],
  "data": Array [
    1,
    1,
  ],
  "labels": Array [
    "Failed",
    "Pending",
  ],
}
`;

exports[`Render tests should create link to coverage report if in config 1`] = `"/some/place/index.html"`;

exports[`Render tests should not create link to coverage report if not in config 1`] = `"#"`;
